provider/pkg/provider/provider.go

```diff
@@ -2201,14 +2201,25 @@ func (k *kubeProvider) serverSidePatch(
 	var newObject *unstructured.Unstructured
 	_, err = client.Get(context.TODO(), newInputs.GetName(), metav1.GetOptions{})
 	switch {
-	case err == nil:
-		newObject, err = client.Patch(context.TODO(), newInputs.GetName(), patchType, patch, metav1.PatchOptions{
+	case errors.IsNotFound(err):
+		newObject, err = client.Create(context.TODO(), newInputs, metav1.CreateOptions{
 			DryRun: []string{metav1.DryRunAll},
 		})
-	case errors.IsNotFound(err):
+	case newInputs.GetNamespace() != oldInputs.GetNamespace():
+		client, err := k.clientSet.ResourceClient(newInputs.GroupVersionKind(), newInputs.GetNamespace())
+		if err != nil {
+			return nil, nil, err
+		}
 		newObject, err = client.Create(context.TODO(), newInputs, metav1.CreateOptions{
```

Comment:
I'm just trying to wrap my head around this... What happens with the object in the old namespace? Don't we need to delete it?

Reply:
This method just computes the patch. The resource replacement logic happens separately from this.